### ISSUE TITLE
reassigning props with `mergeProps` reactivity eslint error

### ISSUE THREAD

**Issue by Tronikelis** [solidjs-community/eslint-plugin-solid#132]
**Your Question**

Hello, I have a question about this specific use-case

```tsx
function C(props) {
// The reactive variable 'props' should not be reassigned or altered directly
 props = mergeProps({d: 1}, props)
}
```

solid plugin catches this behavior as an error, but it works as you can see in this playground https://playground.solidjs.com/anonymous/20849471-1697-4f58-9f6a-c27e1924e31f

should this be allowed?

**Comment by joshwilsonvu** [comment]
This case technically works fine, but reassigning a reactive variable is very difficult for a plugin to analyze. In accordance with many of Solid's examples, we push users towards declaring any reactive variables with `const`. Your component can take a parameter called `_props` and assign the merged props to a constant called `props`--that's what the `no-destructure` rule does. 

The behavior won't change here, but the error message could be improved if you have any suggestions.
